### PR TITLE
Add support for lsp type definitions

### PR DESCRIPTION
**Pull request by elkowar**
This PR adds support for the textDocument/typeDefinition feature, allowing you to jump to the definition of the type of a given value

### CODE REVIEW

**Comment by elkowar** [comment]
Done!

File: lua/trouble/providers/lsp.lua [hunk 1]
```diff
@@ -81,6 +81,27 @@ function M.definitions(win, buf, cb, _options)
   end)
 end
 
+---@return Item[]
+function M.type_definitions(win, buf, cb, _options)
+  local method = "textDocument/typeDefinition"
+  local params = util.make_position_params(win, buf)
+  lsp.buf_request(buf, method, params, function(err, result)
```
Comment (folke) [inline lua/trouble/providers/lsp.lua:88]: this should be `lsp_buf_request` instead of `lsp.buf_request`, so we can make it bakward compatible with 0.5

**Review by folke — CHANGES_REQUESTED** [review]

File: lua/trouble/providers/lsp.lua (hunk 1 above)
Comment (elkowar) [reply lua/trouble/providers/lsp.lua]: missed that, thanks!

**Comment by Saecki** [comment]
Is there anything missing here? Because I would really like to use this feature :upside_down_face:

**Review by folke — APPROVED** [review]
Thank you!

**Pull request merged**